packages/@aws-cdk/aws-rds/lib/cluster.ts

```diff
@@ -542,23 +542,38 @@ export class DatabaseCluster extends DatabaseClusterNew {
    * Adds the single user rotation of the master password to this cluster.
    */
   public addRotationSingleUser(options: RotationSingleUserOptions = {}): secretsmanager.SecretRotation {
+    if (!options.secret || options.secret === this.secret) {
+      return this.addSingleUserRotationForMasterSecret(options);
+    }
+
+    const id = 'RotationSingleUser';
```

Comment:
Let's inline this variable.

Reply:
Shouldn't we use a_different_ ID than what `addSingleUserRotationForMasterSecret` uses?

Reply:
Since the scope is different (than `this`) there's no need, but we can do it if you like. BTW there's a test covering this.